subprojects/snapshots/src/main/java/org/gradle/internal/fingerprint/impl/NormalizedPathFingerprintCompareStrategy.java

```diff
@@ -87,18 +87,22 @@ private boolean doVisitChangesSince(
             String normalizedPath = previousFingerprint.getNormalizedPath();
             FileType previousFingerprintType = previousFingerprint.getType();
 
+            DefaultFileChange change;
```

Comment:
The type of these local variables, and also the return types of the `modified()` etc. methods should be `FileChange` only.

Reply:
Or just `Change`, since that's what `visitChange` requires anyway, thanks, fixed.